src/plugins/workspace/server/plugin.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import {
+  PluginInitializerContext,
+  CoreSetup,
+  Plugin,
+  Logger,
+  CoreStart,
+} from '../../../core/server';
+import { IWorkspaceDBImpl } from './types';
+import { WorkspaceClientWithSavedObject } from './workspace_client';
+import { registerRoutes } from './routes';
+
+export class WorkspacePlugin implements Plugin<{}, {}> {
+  private readonly logger: Logger;
+  private client?: IWorkspaceDBImpl;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get('plugins', 'workspace');
+  }
+
+  public async setup(core: CoreSetup) {
+    this.logger.debug('Setting up Workspaces service');
+
+    this.client = new WorkspaceClientWithSavedObject(core);
+
```

Comment:
I believe somewhere you will want to register your saved object type of `workspace` so we can export that new saved object type if needed.

Reply:
Thanks kavilla but we can export `workspace` in later PRs if it is needed, for this PR, we can keep as less exports as possbile.

Reply:
Gotcha, is there a meta issue or board that is keeping track of requirements?

I think we definitely still want to have the ability to export the workspace. Everything here https://playground.opensearch.org/app/management/opensearch-dashboards/objects is saved within the OSD index and exportable. If we export it out without workspaces but the other saved objects have the context of a workspace but the workspaces.

Reply:
Sorry I realized that maybe I was little bit unclear with my first comment. So just in case, doing something like this part:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/vis_builder/server/plugin.ts#L29

Reply:
Sure, we are doing the register [here](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5075/files#diff-38cb975b307e4ac766eed69f790e0312a0a08411a1d05960cfbbc9b951e6b99aR59)